detox/ios/Detox/Invocation/Action.swift

```diff
@@ -435,8 +435,22 @@ class ScrollToEdgeAction : Action {
 			break;
 		}
 		
-		element.scroll(to: targetEdge)
-		
+		let startPositionX : Double
+		if params?.count ?? 0 > 1, let param1 = params?[1] as? Double, param1.isNaN == false {
+			startPositionX = param1
+		} else {
+			startPositionX = Double.nan
+		}
+		let startPositionY : Double
+		if params?.count ?? 0 > 2, let param2 = params?[2] as? Double, param2.isNaN == false {
+			startPositionY = param2
+		} else {
+			startPositionY = Double.nan
+		}
```

Comment:
We should extract this repetitive code to a function (outside of this class so it will be reused in `ScrollAction` as well
```
func startPosition(forIndex index: Int, in params: [Any]?) -> Double {
    guard params?.count ?? 0 > index, 
          let param = params?[index] as? Double, 
          param.isNaN == false else {
        return Double.nan
    }
    return param
}
```

Then use it this way:
```swift
let startPositionX = startPosition(forIndex: 1, in: params)
let startPositionY = startPosition(forIndex: 2, in: params)
```